File: core-algorithms/1_recursion/2_fileSearcher/fileSearcher.py
import json

with open("./files.json", mode="r") as f:
    files = json.load(f)

def fileSearcher(fs, allFiles = []):

    if(not isinstance(fs, dict)):
        return
    
    for key, value in fs.items():
        if(key == "files"):
            allFiles.extend(value)
        else:
            fileSearcher(value, allFiles)
    
    return allFiles

print(fileSearcher(files))

File: core-algorithms/1_recursion/2_fileSearcher/fileSearcher.ts
import files from "./files.json" with { type: "json" };

const fileSearcher = (data: any, files: string[] = []) => {
  if (!data || typeof data !== "object") return files;

  for (let i in data) {
    if (i === "files" && Array.isArray(data[i])) {
      files.push(...data[i]);
    } else {
      files = fileSearcher(data[i], files);
    }
  }

  return files;
};

console.log(fileSearcher(files));
